### PR TITLE
Mice and Regal Rats won't spawn in the icebox solar panels

### LINKED ISSUE #73238: Feral Regal Rats Spawning in the cold
Reporting client version: 514.1589

## Round ID:
[199534](https://scrubby.melonmesa.com/round/199534)

## Reproduction:

Round starts, there is already a dead body, orbit dead body to see how they died so fast. Is ghost controllable regal rat that spawned in Icebox's South East solars, outside, spawned in a couple rats, then succumbed to the hostile atmos that is the outside of icebox. I've been told that apparently the parameters for spawning a feral regal rat is as follows, courtesy of Spook,
```dm
mouse = new /mob/living/simple_animal/hostile/regalrat/controlled(proposed_turf)
        if(proposed_turf.air.has_gas(/datum/gas/oxygen, 5))
            num_mice -= 1
            mouse = null
```
Which you know, doesn't check for cold n such.

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Fixes #73238

This adds a temperature check to the turf locator in the mice migration event.
Also it refactors the check a bit because its loop seemed... odd? If there was some reason the old version was more efficient let me know but I can't see a purpose to do it that way around.
Also then the same numbers were being used in 3 separate files so I moved it into a define.
_Finally_ I removed a redundant element from the cockroach while I was looking at the files.

There was a moment when I started looking at this where I thought "I should add a generic helper proc which is given a turf and a mob and can tell you if it would die if it spawned on that turf".
Then I looked at lungs and immediately decided that was going to be a massive amount of work for this trivial bug fix... maybe one day.

## Why It's Good For The Game

This was sparing mappers from the punishment whoever added this feature surely intended for them to suffer.
Also it's just a bit sad to notify ghosts to join a ghost role which will die instantly 😢

## Changelog

:cl:
fix: Mice and Regal Rats will no longer spawn in icebox's solar panels and die of cold.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/minor_mapping.dm [hunk 1]
```diff
@@ -16,26 +16,20 @@ SUBSYSTEM_DEF(minor_mapping)
...
-
 	while((num_mice > 0) && exposed_wires.len)
 		proposed_turf = pick_n_take(exposed_wires)
-
-		if(!istype(proposed_turf))
+		if (!istype(proposed_turf)||
+			!proposed_turf.air.has_gas(/datum/gas/oxygen, 5) ||
+			proposed_turf.temperature > NPC_DEFAULT_MAX_TEMP ||
+			proposed_turf.temperature < NPC_DEFAULT_MIN_TEMP)
 			continue
```
Comment (ZephyrTFA) [inline code/controllers/subsystem/minor_mapping.dm:26]:
```suggestion
		if (!istype(proposed_turf) || \
			!proposed_turf.air.has_gas(/datum/gas/oxygen, 5) || \
			proposed_turf.temperature > NPC_DEFAULT_MAX_TEMP || \
			proposed_turf.temperature < NPC_DEFAULT_MIN_TEMP)
			continue
```

Comment (ZephyrTFA) [reply code/controllers/subsystem/minor_mapping.dm:26]: I think byond doesn't know how to parse `... || \n`

Comment (Jacquerel) [reply code/controllers/subsystem/minor_mapping.dm:26]: Does it need the backslashes? If statements can natively support being multilined, I thought those should only be required if it was a macro.

File: code/controllers/subsystem/minor_mapping.dm [hunk 2]
```diff
@@ -16,26 +16,20 @@ SUBSYSTEM_DEF(minor_mapping)
...
-
-
 	while((num_mice > 0) && exposed_wires.len)
 		proposed_turf = pick_n_take(exposed_wires)
-
-		if(!istype(proposed_turf))
+		if (!istype(proposed_turf) || \
+			!proposed_turf.air.has_gas(/datum/gas/oxygen, 5) || \
+			proposed_turf.temperature > NPC_DEFAULT_MAX_TEMP || \
+			proposed_turf.temperature < NPC_DEFAULT_MIN_TEMP)
```
Comment (san7890) [inline code/controllers/subsystem/minor_mapping.dm:25]: this is ugly to me (wack indentation for something like this IMO), just spin it out into it's own proc that returns TRUE/FALSE depending on turf validity like

```txt
ssminormapping/proc/valid_turf(turf/open/checkable)
    if(!istype(checkable)
        return FALSE
    var/turf_gasmix = checkable.air // (this one is optional actually)
    var/temperature = checkable.temperature // (this one makes good logical sense we save 0.00001 CPU cycles with this cache
     if(...)
```

don't have to do it exactly the way i did it but this is much cleaner to me footprint-wise.

File: code/controllers/subsystem/minor_mapping.dm [hunk 3]
```diff
@@ -16,26 +16,20 @@ SUBSYSTEM_DEF(minor_mapping)
...
 			continue
 
-		if(prob(PROB_MOUSE_SPAWN))
-			if(!mouse)
-				mouse = new(proposed_turf)
-			else
-				mouse.forceMove(proposed_turf)
+		num_mice -= 1
```
Comment (san7890) [inline code/controllers/subsystem/minor_mapping.dm:28]: does a decrement operator not work here like `num_mice--` or is that just not trustable for whatever reason (i know you just transplanted the code from one place to here but i'm just curious why this is)

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/minor_mapping.dm (hunk 3 above)
Comment (Jacquerel) [reply code/controllers/subsystem/minor_mapping.dm:28]: I don't even know why the code was originally written in the way it was, seemed a pretty roundabout way to do it. If I had noticed this I would have changed it.

Comment (san7890) [reply code/controllers/subsystem/minor_mapping.dm:28]: that's fine just wondering if you noticed it and didn't change it deliberately

**Pull request merged**